tfidf/preprocess.py

```diff
@@ -158,7 +183,7 @@ def normalize_term(self, text):
         text = self.handle_stopwords(text)
         return self.stem_term(text)
 
-    @lru_cache(maxsize=10000)
+    @cached(LRUCache(maxsize=10000))
```

Comment:
Thought for the future - a pluggable cacheing system. LRUCache is pretty good in most cases but if the client can specify which cache to implement, it increases scaleability quite a bit.